src/components/BooleanInput/index.ts

```diff
@@ -27,8 +27,9 @@ class BooleanInput extends Input implements IBindable, IFocusable {
     protected _value: boolean;
 
     constructor(args: BooleanInputArgs = {}) {
-        args.tabIndex = args.tabIndex ?? 0;
-        super(args.dom, args);
+        args.tabIndex ??= 0;
```

Comment:
Wondering whether destructuring directly and leaving the original args might be better?

Reply:
I was wondering the same thing myself. But in the case of `tabIndex`, the code was updating the supplied options object before this PR. This PR does not also update the supplied options object's dom property in some places now though. Hmm, I guess technically that's not great....

Reply:
So you'd prefer this?

```
        if (args.tabIndex === undefined) {
            args = {
                ...args,
                tabIndex: 0
            };
        }
```

Reply:
Either that or just:
`super({ tabIndex: 0, …tags })`

Reply:
But the user might have set `tabIndex` themselves. Your suggestion would always ignore the user-supplied value, no?

Reply:
I've updated the code. There should no longer be any instances where the supplied options object is modified.

Reply:
Doing:
```
super({ tabindex: 0, ...args });
```
will result in `tabIndex` being overridden by args if it has it present.

Reply:
Oh yeah, cool! So I've updated the code again to use this where applicable. Thanks!